File: src/main/resources/output/IFC4X3_python/Entities/IfcProfileDef/IfcProfileDef.py
class IfcProfileDef:
    def __init__(self):
        self.ProfileType: IfcProfileTypeEnum = None
        self.ProfileName: IfcLabel = None

    # === EXTENDED BY ===
    # IfcArbitraryClosedProfileDef
    # IfcArbitraryOpenProfileDef
    # IfcCompositeProfileDef
    # IfcDerivedProfileDef
    # IfcOpenCrossProfileDef
    # IfcParameterizedProfileDef

    # === INVERSE CLAUSES ===
    # HasExternalReference : SET [0:?] OF IfcExternalReferenceRelationship FOR RelatedResourceObjects
    # HasProperties : SET [0:?] OF IfcProfileProperties FOR ProfileDefinition

File: src/main/resources/output/IFC4X3_vbnet/Entities/IfcProfileDef/IfcProfileDef.vb
Public Class IfcProfileDef
    Public ProfileType As IfcProfileTypeEnum
    Public ProfileName As IfcLabel

    ' === EXTENDED BY ===
    ' IfcArbitraryClosedProfileDef
    ' IfcArbitraryOpenProfileDef
    ' IfcCompositeProfileDef
    ' IfcDerivedProfileDef
    ' IfcOpenCrossProfileDef
    ' IfcParameterizedProfileDef

    ' === INVERSE CLAUSES ===
    ' HasExternalReference : SET [0:?] OF IfcExternalReferenceRelationship FOR RelatedResourceObjects
    ' HasProperties : SET [0:?] OF IfcProfileProperties FOR ProfileDefinition
End Class
